src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -1359,6 +1376,70 @@ def tag(self, tags):
+        from msrestazure.azure_operation import AzureOperationPoller
+        query_parameters = {}
+        serialize = self.rcf.resources._serialize  # pylint: disable=protected-access
+        deserialize = self.rcf.resources._deserialize  # pylint: disable=protected-access
+        client = self.rcf.resources._client  # pylint: disable=protected-access
+
+        if not url:
+            url = '/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/' \
+                '{resourceProviderNamespace}/{parentResourcePath}/{resourceType}/{resourceName}/{action}'
+            path_format_arguments = {
+                'resourceGroupName': serialize.url(
+                    "resource_group_name", self.resource_group_name, 'str',
+                    max_length=90, min_length=1, pattern=r'^[-\w\._\(\)]+$'),
+                'resourceProviderNamespace': serialize.url(
+                    "resource_provider_namespace", self.resource_provider_namespace, 'str'),
+                'parentResourcePath': serialize.url(
+                    "parent_resource_path", self.parent_resource_path, 'str', skip_quote=True),
+                'resourceType': serialize.url("resource_type", self.resource_type, 'str', skip_quote=True),
+                'resourceName': serialize.url("resource_name", self.resource_name, 'str'),
+                'subscriptionId': serialize.url(
+                    "self.config.subscription_id", self.rcf.resources.config.subscription_id, 'str'),
+                'action': serialize.url("action", action, 'str')
+            }
+            url = client.format_url(url, **path_format_arguments)
+
+            # Construct parameters
+            query_parameters['api-version'] = serialize.query("api_version", self.api_version, 'str')
+
+        # Construct headers
+        header_parameters = {}
+        header_parameters['Content-Type'] = 'application/json; charset=utf-8'
+        if self.rcf.resources.config.generate_client_request_id:
+            header_parameters['x-ms-client-request-id'] = str(uuid.uuid1())
+        if self.rcf.resources.config.accept_language is not None:
+            header_parameters['accept-language'] = serialize.header(
+                "self.config.accept_language", self.rcf.resources.config.accept_language, 'str')
+
+        # Construct and send request
+        def long_running_send():
+            request = client.post(url, query_parameters)
+            return client.send(
+                request, header_parameters, json.loads(parameters) if parameters else None)
+
+        def get_long_running_status(status_link, headers=None):
+            request = client.get(status_link)
+            if headers:
+                request.headers.update(headers)
+            return client.send(request, header_parameters)
+
+        def get_long_running_output(response):
+            from msrestazure.azure_exceptions import CloudError
+            if response.status_code not in [200, 202, 204]:
```

Comment:
If we will cover non-post action, let us cross check to make sure we include more valid http return codes

Reply:
Shouldn't all status code between 200 and 299 acceptable?

Reply:
According to ARM this is enough:
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#call-action-post